framework/src/main/java/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java

```diff
@@ -364,6 +365,28 @@ public void standardizeVariable(Tree node, AnnotatedTypeMirror type, Element ele
         if (path == null) {
             return;
         }
+
+        if (ele instanceof DetachedVarSymbol) {
+            // Skip standardizing artificial variable trees, because it is the standardized
+            // variable name that is used to construct and initialize an artificial variable.
+            // For example,
+            //
+            //    static HashMap<Integer, Date> call_hashmap = new HashMap<>();
+            //
+            //    public foo() {
+            //        for (Integer i : call_hashmap.keySet()) {
+            //            @NonNull Date d = call_hashmap.get(i);
+            //        }
+            //    }
+            //
+            // The CFGBuilder creates the initialized artificial iterator as follows:
+            //      iter#num0 = NonNullMapValue.call_hashmap.keySet().iterator()
```

Comment:
The `NonNullMapValue` comes as a surprise. I assume that's the class around the above code? Maybe add that there.